### PR TITLE
Add counter:reset method

### LINKED ISSUE #260: Counter initialization
It is implied that counter start from 0, since counter value after its first `inc(1)` will be `1`.

But metric value do not exist until first `inc()` call (since there are no other methods to change counter metric value). So if value if never changed, metric will be null.

For example, it is a common case to count process errors with counter. But if we did not have any errors from the start, the metric will be null. HTTP activity is the same (https://github.com/tarantool/metrics/issues/96): it is more convenient to know that we had `0` requests on some route rather then `no data`. `No data` if often treated like "there is something wrong with metric retrieval" rather then "it's a plain zero".

It is now possible to initialize counter with specific labels with
```lua
counter:inc(0, label_pairs)
```
but it is definitely a duct tape. It would be better to support this behavior explicitly, e.g. with some collector method.

### PR DESCRIPTION
**Pull request by yngvar-antonsson**
I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation (README and rst)

Close #260

### CODE REVIEW

File: metrics/collectors/counter.lua [hunk 1]
```diff
@@ -9,4 +9,8 @@ function Counter:inc(num, label_pairs)
     Shared.inc(self, num, label_pairs)
 end
 
+function Counter:reset(label_pairs)
```
Comment (DifferentialOrange) [inline metrics/collectors/counter.lua:12]: The original request was about counter initialization. Initialization with `counter:reset(label_pairs)` seems kinda weird. How about adding an alias for `reset` as a little syntactic sugar?

```lua
Counter.initialize = reset
```

**Review by DifferentialOrange — COMMENTED** [review]
PR's missing changelog and doc edits

File: metrics/collectors/counter.lua (hunk 1 above)
Comment (rosik) [reply metrics/collectors/counter.lua:12]: init / reinit / reset, who cares. Reset is better than inc(0), and that's the purpose.

Comment (rosik) [reply metrics/collectors/counter.lua:12]: Init which resets the value may be considered weird too.

**Review by DifferentialOrange — APPROVED** [review]
Well, I'm still prefer having both `init` and `reset`, but if you're against it, let's at least add the functionality

**Pull request merged**